sphinx/util/osutil.py

```diff
@@ -98,6 +98,10 @@ def walk(top, topdown=True, followlinks=False):
     """Backport of os.walk from 2.6, where the *followlinks* argument was
     added.
     """
+    warnings.warn('sphinx.util.osutil.walk() is deprecated for removal. '
+                  'Please use os.walk() instead.',
+                  RemovedInSphinx40Warning)
+
     names = os.listdir(top)
 
     dirs, nondirs = [], []
```

Comment:
Could we just replace all of this with a call to `os.walk`? It should be a separate change but in theory, they do the exact same thing now, right? That means we could delete any tests that might exist for this too.